### PR TITLE
Upgrade test-snaps to use Node v16

### PR DESCRIPTION
**Pull request by bowensanders**
Addresses issue #51 

Changes made to .nvmrc and to individual package.json files
- .nvmrc now points to v16
- package.json engines section now require >=16.15.1

### CODE REVIEW

**Comment by bowensanders** [comment]
Note: Right now, unlike Frederik's PR (which i mistakenly made this one myself anyway, but it made me do all the things myself which was a learning experience) I am only checking vs. node v16 - not v16 & 18 - the v18 check is broken in #51 

Also instead of >=16.0.0 I set it to the version we're using which is >=16.15.1

If this is not the preferred method, please let me know

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:
-        node-version: [14.x, 16.x]
+        node-version: [16.x]
```
Comment (rekmarks) [inline .github/workflows/build-test.yml:14]: Future proofing:

```suggestion
        node-version: [16.x, 18.x]
```

File: package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
   "private": true,
   "version": "2.0.0",
   "engines": {
-    "node": ">=14.0.0"
+    "node": ">=16.15.1"
```
Comment (rekmarks) [inline package.json:6]: We should keep this to the smallest version possible. The same goes for all the 'engines' fields in all `package.json` files.
```suggestion
    "node": ">=16.0.0"
```

**Review by rekmarks — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (bowensanders) [reply package.json]: requested changes made :)

**Review by rekmarks — APPROVED** [review]

**Comment by rekmarks** [comment]
The logic behind doing `>=16.0.0` instead of `>=16.x.y` is that we only care about the major version, not the minor or patch versions. People may have different versions of 16.x installed locally and that's fine.

**Pull request merged**